ocs_ci/utility/utils.py

```diff
@@ -5316,3 +5316,55 @@ def create_config_ini_file(params):
     log.info(f"config.ini file for cluster is located at {config_ini_file.name}")
 
     return config_ini_file.name
+
+
+def generate_folder_with_files(num_files: int = 300) -> str:
+    """
+    Generates a random folder with specified number of random text files inside it in /tmp folder.
+
+    Args:
+        num_files (int): Number of files to generate. Defaults to 300.
+
+    Returns:
+        str: Full path to the generated folder.
+
+    Raises:
+        OSError: If folder creation or file write fails.
+        RuntimeError: If file count validation fails.
+    """
+
+    def _generate_files():
+        for _ in range(num_files):
+            filename = (
+                "".join(random.choices(string.ascii_lowercase + string.digits, k=8))
+                + ".txt"
+            )
+            filepath = os.path.join(folder_path, filename)
+
+            content = "".join(
+                random.choices(string.ascii_letters + string.digits + " \n", k=100)
+            )
+
+            with open(filepath, "w") as f:
+                f.write(content)
+                f.flush()
+                os.fsync(f.fileno())  # Ensure data is written to disk
+
+            yield filepath
```

Comment:
It is suggested do not use yeild with pytest. pytest does not recognize generator functions and it may brake execution flow. 
Maybe I am mistaken here, @petr-balogh, please check my statement.

Reply:
I am not aware of such restriction @DanielOsypenko .

But even the generator is used here - it's usage doesn't makes much sense as:
The generator _generate_files() is only iterated once when set(_generate_files()) is created, meaning all file paths are generated and stored in memory at once. This eliminates the benefit of using a generator.